### PR TITLE
fix: change  string concat

### PR DESCRIPTION
**Pull request by gy09535**
### What this PR does / why we need it:

fix:https://github.com/apache/apisix/issues/2930
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/sls-logger.lua [hunk 1]
```diff
@@ -126,17 +127,13 @@ local function remove_stale_objects(premature)
...
-        end
-
-        data = data .. entry.data
+        table.insert(data, entry.data)
         core.log.info("buffered logs:", entry.data)
     end
 
-    return data
+    return table.concat(data)
 end
```
Comment (spacewander) [inline apisix/plugins/sls-logger.lua:137]: Better to put `_M.combine_syslog = combine_syslog` here.

Comment (gy09535) [reply apisix/plugins/sls-logger.lua:137]: fixed.

File: apisix/plugins/sls-logger.lua [hunk 2]
```diff
@@ -126,26 +127,23 @@ local function remove_stale_objects(premature)
...
 
 local function combine_syslog(entries)
-    local data
+    local data = {}
```
Comment (membphis) [inline apisix/plugins/sls-logger.lua:130]: how about the name `items`?

Comment (gy09535) [reply apisix/plugins/sls-logger.lua:130]: Good, I am tangled what name for the local var.

Comment (gy09535) [reply apisix/plugins/sls-logger.lua]: fixed.

**Review by spacewander — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**